torch/onnx/symbolic.py

```diff
@@ -446,6 +446,15 @@ def leaky_relu(g, input, negative_slope, inplace=False):
     return g.op("LeakyRelu", input, alpha_f=_scalar(negative_slope))
 
 
+def linear(g, input, weight, bias=None):
+    if input.dim() == 2 and bias is not None:
```

Comment:
`input`, `weight` and `bias` here should be instances of `Value` from ir.h. 

Then to get sizes, do something like this: https://github.com/pytorch/pytorch/blob/master/torch/onnx/symbolic.py#L379 Note you need a check that the specified Value is of TensorType. Only then will sizes be specified, otherwise, you'll need to emit a generic, rank-invariant version.

Btw I'm not sure if `bias` will be none here, can you try it out passing tensor() as the third argument to aten::linear and tracing it?

Reply:
1. I assume `@parse_args('v', 'v', 'v')` will do this?
2. > you'll need to emit a generic, rank-invariant version
    That's the second part of this function right?
3. > Btw I'm not sure if bias will be none here, can you try it out passing tensor() as the third argument to aten::linear and tracing it?
    Can you be more explicit in what you want me try here? Do you mean Python or C++? Do you mean `at::linear`?

I have no idea what this symbolic stuff is, so please bear with me